### PR TITLE
Enable deprecation warnings for Django 1.11 apps.

### PR DESCRIPTION
**Pull request by doctoryes**
Configuration Pull Request
---
https://openedx.atlassian.net/browse/PLAT-1350

@feanil I don't necessarily think this change is the best way to go about turning on these warnings in production Django apps - but it's similar to the approach in this edx-platform PR:

https://github.com/edx/edx-platform/pull/16799

So I made this PR to start the conversation. Can you suggest a better way to do this?

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by doctoryes** [comment]
@edx/devops Ping - just wanting to know what you think about this approach to enable warnings in production edxapp services.

File: playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2 [hunk 1]
```diff
@@ -5,9 +5,9 @@
 {% set edx_django_service_venv_bin = edx_django_service_venv_dir + "/bin" %}
 
 {% if COMMON_ENABLE_NEWRELIC_APP %}
-{% set executable = edx_django_service_venv_bin + '/newrelic-admin run-program ' + edx_django_service_venv_bin + '/gunicorn' %}
+{% set executable = ' -Wd ' + edx_django_service_venv_bin + '/newrelic-admin run-program ' + edx_django_service_venv_bin + '/gunicorn' %}
```
Comment (feanil) [inline playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: I think this is in the wrong place.  I think you want this after `gunicorn` right?

Comment (feanil) [reply playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: I think you also want to do it here: https://github.com/edx/configuration/blob/master/playbooks/roles/edxapp/templates/lms.conf.j2
and here: https://github.com/edx/configuration/blob/master/playbooks/roles/edxapp/templates/cms.conf.j2

Comment (doctoryes) [reply playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: Thanks! I've added the flag there as well.

**Comment by doctoryes** [comment]
@fredsmith @feanil Comments have been addressed. Re-review?

File: playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2 (hunk 1 above)
Comment (feanil) [reply playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: @doctoryes I think we still need to move `-Wd` to after the call to `gunicorn` call right?  What script are these flags for?

Comment (doctoryes) [reply playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: @feanil On a sandbox, I'm seeing command lines like below for non-LMS/CMS IDAs:
```
/edx/app/xserver/venvs/xserver/bin/python /edx/app/xserver/venvs/xserver/bin/gunicorn -c /edx/app/xserver/xserver_gunicorn.py pyxserver_wsgi:application
/edx/app/credentials/venvs/credentials/bin/python3 /edx/app/credentials/venvs/credentials/bin/gunicorn -c /edx/app/credentials/credentials_gunicorn.py credentials.wsgi:application
/edx/app/ecommerce/venvs/ecommerce/bin/python2.7 /edx/app/ecommerce/venvs/ecommerce/bin/gunicorn -c /edx/app/ecommerce/ecommerce_gunicorn.py ecommerce.wsgi:application
```
The `-Wd` needs to go between the Python executable and the gunicorn call. So it *seems* like this the correct place? I'm less sure about the New Relic invocation - I'm assuming that an absolute Python path is the first thing on that command-line as well.

File: playbooks/roles/edxapp/templates/cms.conf.j2 [hunk 2]
```diff
@@ -1,9 +1,9 @@
 [program:cms]
 
 {% if COMMON_ENABLE_NEWRELIC_APP -%}
-{% set executable = edxapp_venv_dir + '/bin/newrelic-admin run-program ' + edxapp_venv_dir + '/bin/gunicorn' -%}
+{% set executable = ' -Wd ' + edxapp_venv_dir + '/bin/newrelic-admin run-program ' + edxapp_venv_dir + '/bin/gunicorn' -%}
 {% else -%}
-{% set executable = edxapp_venv_dir + '/bin/gunicorn' -%}
+{% set executable = ' -Wd ' + edxapp_venv_dir + '/bin/gunicorn' -%}
```
Comment (doctoryes) [inline playbooks/roles/edxapp/templates/cms.conf.j2]: @feanil I don't think the changes in this file apply to the workers. Where would I change configuration to enable warnings for LMS/CMS workers as well?

Comment (doctoryes) [reply playbooks/roles/edxapp/templates/cms.conf.j2]: Here's their command-line that I see on a sandbox:
```
/edx/app/edxapp/venvs/edxapp/bin/python /edx/app/edxapp/edx-platform/manage.py cms --settings=aws celery worker --loglevel=info --queues=edx.cms.core.default --hostname=edx.cms.core.default.%h --concurrency=1
```

File: playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2 (hunk 1 above)
Comment (feanil) [reply playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: I'm not sure where you see that, I'm looking at `/edx/app/supervisor/conf.d/lms.conf` where I see.

`command=/edx/app/edxapp/venvs/edxapp/bin/gunicorn -c /edx/app/edxapp/lms_gunicorn.py lms.wsgi`

Comment (doctoryes) [reply playbooks/roles/edx_django_service/templates/edx/app/app/app.sh.j2:8]: @feanil I got those lines from a `ps aux` listing on a sandbox. But I'm thinking I was misled. I'm going to put the flag after the gunicorn as you suggest and try to build a sandbox.

**Comment by doctoryes** [comment]
I tried these changes out on a sandbox - and they didn't work at all. We'll likely try to enable these warnings at run-time instead.

**Pull request closed without merging**